### DISCUSSION TITLE
Add option(s) to disable markdown shortcuts

### DISCUSSION THREAD

**Discussion by labkey-alan** [remirror/remirror#1526] [Ideas] [answered]
Hello,

We're using remirror to render an editor and our users are trying to input text that looks like this:

> file_path = /home/user/stuff/my_file_name.txt

Remirror has input rules in the ItalicExtension that remove these underscores and renders the word "file" in italics. In general we would like to disable all of these markdown-like input rules, but there is no way to do that.

Would it be possible to add a configuration option to the various extensions (Bold, Italic, etc.) to disable these input rules, and paste rules?

**Comment by labkey-alan** [comment]
Here is what we went with in the mean time:

```typescript
export class MyBoldExtension extends BoldExtension {
    createInputRules(): InputRule[] {
        return [];
    }
}

export class MyItalicExtension extends ItalicExtension {
    createInputRules(): InputRule[] {
        return [];
    }

    createPasteRules(): MarkPasteRule[] {
        return [];
    }
}

export class MyHeadingExtension extends HeadingExtension {
    createInputRules(): InputRule[] {
        return [];
    }

    createPasteRules(): NodePasteRule[] {
        return [];
    }
}
```

Ideally we wouldn't need to do this, and instead would do something like this:

```typescript
new BoldExtension({ disableMarkdownRules: true });
```

**Comment by labkey-alan** [comment]
@whawker you mentioned on [Twitter](https://twitter.com/will_hawker/status/1536095354453639169) that I could use this:

```
useRemirror({
  extensions: () => [],
  settings: {
    exclude: { inputRules: true }
  }
})
```

Does that disable all inputRules across all extensions though? In my case I only want to disable the markdown rules in the Bold, Italic, and Heading extensions. There are other input rules that we use.

Thanks!

**Comment by whawker** [comment, marked as answer]
@labkey-alan Yes, that would disable across all extensions. To disable specific extensions you could

```js
useRemirror({
  extensions: () => [
    new BoldExtension({ exclude: { inputRules: true } }),
    new ItalicExtension({ exclude: { inputRules: true } }),
    new HeadingExtension({ exclude: { inputRules: true } }),
  ],
})
```
